### ISSUE TITLE
how to run the app 

### ISSUE THREAD

**Issue by paulakhalil** [iyaja/llama-fs#19]
i did pip install -r requirements.txt successfully. now what? how do i run the app?

**Comment by bzanghi** [comment]
I can't fully get it working, but a good start would be:
1. In the terminal: fastapi dev server.py
2. Go to the API docs: https:/127.0.0.1:8000/docs
3. On the "batch" route update the parameters
     I.e. Try it out > update the path 
     Update incognito=true (if you want to use ollama)

**Comment by BrainRTP** [comment]
It seems time to do a normal description of the project startup process with use cases

**Comment by OtenMoten** [comment]
Please create a tutorial and graphical user interface!

**Comment by Enferlain** [comment]
Yeah I don't really get how this has 3.5k stars but barely any instructions for achieving the use case. And "ollama pull moondream" doesn't work on windows

**Comment by vnl** [comment]
Set Up the GUI Application

1. Navigate to the Electron React app directory:
    `cd electron-react-app  `  
2. Ensure you have npm (Node Package Manager) installed. 
3. Install the necessary dependencies:
   ` npm install`
4. Start the desktop application:
    `npm start`
    
I am having issues running the Electron app, but I am hoping others have better luck.

**Comment by robeyl** [comment]
> Yeah I don't really get how this has 3.5k stars but barely any instructions for achieving the use case. And "ollama pull moondream" doesn't work on windows

Same on MacOS: "command not found: ollama"

**Comment by BarbokOL** [comment]
> Yeah I don't really get how this has 3.5k stars but barely any instructions for achieving the use case. And "ollama pull moondream" doesn't work on windows

I'm just as confused, the instructions are there but they just don't really make a lot of sense, granted im a complete noob but still

**Comment by Japillow** [comment]
On MacOS, using the current version of Node, i.e. Node 22, I get 
```
$ npm install
npm error Invalid property "node"
```
Reverting to Node 18 (using for ex `brew install node@18` fixes it for me

**Comment by EmJayArr** [comment]
Here's what I had to do to get everything at least running:

1. In ./requirements.txt: change "agentops" to "agentops==0.3.26" and add "fastapi[standard]" at the end.
2. In ./electron-react-app/package.json:  somewhere (I did it on line 3 underneath "description") add: 
`  "type": "json",` 
3. In package.json again, change devEngines to engines:
```
 "engines": {
    "node": ">=14.x",
    "npm": ">=7.x"
  },
```

4. Groq has deprecated llama-3.1-70b. Replace all occurrences of it with llama-3.3-70b. From the linux terminal, cd to the directory and then run something like this:
`find . -type f -exec sed -i 's/llama-3.1-70b/llama-3.3-70b/g' {} +` 

5. I created a python virtual environment:
```
$ python3 -m venv ./.venv
$ source .venv/bin/activate
$ pip install nodeenv
$ nodeenv -p
$ source .venv/bin/activate # not sure if it's necessary to do this again
```

6. Now we can follow the server instructions:
```
$ pip install -r requirements.txt

$ cp .env.example .env
$ nano .env                      # add your api keys

$ fastapi dev server.py
``` 

7. Then to run the GUI, in another terminal tab/window, cd to the electron-react-app directory:
```
$ cd electron-react-app
$ npm install
$ npx update-browserslist-db@latest    # gets rid of a warning
$ npm start
```

This got me to where I could get a suggestion, but the "Confirm selected changes" button didn't actually create the folders and move the files or anything.

Also, for those wondering about moondream I've installed ollama separately, but I don't think that this is actually set up to work in incognito mode yet either.
